--- /dev/null	2022-06-25 09:39:27.249629944 -0400
+++ source/t/TestMonkeyBarrelDemo.mc	2022-06-26 09:45:09.825932834 -0400
@@ -0,0 +1,31 @@
+//
+//SPDX-FileCopyrightText: 2021-2022 Wesley Schwengle <wesley@opperschaap.net>
+//
+//SPDX-License-Identifier: BSD-3-Clause
+//
+using Toybox.Test as t;
+import Toybox.Lang;
+using Toybox.System as Sys;
+
+using MonkeyBarrelDemo.demo as demo;
+
+module MonkeyBarrelDemo {
+  module TestMonkeyBarrelDemo {
+
+    (:test)
+    function testFoo(logger) {
+      t.assertMessage(demo.foo(), "MonkeyBarrelDemo.demo.foo() returns true");
+      return true;
+    }
+
+    (:test)
+    function testBar(logger) {
+      t.assertEqualMessage(
+        demo.bar(),
+        false,
+        "MonkeyBarrelDemo.demo.bar() returns false"
+      );
+      return true;
+    }
+  }
+}
